spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/discovery/KubernetesDiscoveryProperties.java

```diff
@@ -56,14 +58,16 @@ public record KubernetesDiscoveryProperties(
 		@DefaultValue Map<String, String> serviceLabels, String primaryPortName,
 		@DefaultValue Metadata metadata,
 		@DefaultValue("" + DEFAULT_ORDER) int order,
-		boolean useEndpointSlices) {
+		boolean useEndpointSlices,
+		boolean includeExternalNameServices) {
```

Comment:
I added `includeExternalNameServices` only in properties as the actual support for it will be added in the upcoming PR. The fact is, I started working on support for `ExternalName` type of services and realized that unless I clean-up things, searching and filtering services is going to be scattered across multiple places (an adapter I recently introduced and an actual place where external-services support will be added). 

So if its OK, I will leave this flag here (will all the test changes also) and once this is merged, will continue work on ExternalName type of services.